### ISSUE TITLE
Cutelyst built-in to other Qt app

### ISSUE THREAD

**Issue by MikePooh** [cutelyst/cutelyst#305]
I have standalone application written in Qt with some data inside. I need to represent that data in some web interface and give data through REST API. Is that possible to built-in the Cutelyst to another Qt/C++ application to make seamless single app with web server inside? If so, where I can get documentation/examples?

**Comment by dantti** [comment]
yes, you basically need to add Cutelyst::WSGI module or Cutelyst::Server(v3) and setApplication(), but there are still some stuff to fix for the mixed with GUI app (if you have it) that it might exit the app due server errors (like fail to listen and such), I still need to polish this a bit more...

**Comment by MikePooh** [comment]
Yes. My app has GUI interface written in QML.

**Comment by dantti** [comment]
right so link like (Server or WSGI):
https://github.com/cutelyst/cutelyst/blob/master/server/CMakeLists.txt#L102
and use it a bit like main.cpp of the same directory, and please do some PR to help me emit signals instead of qFatal or exit :)

**Comment by MikePooh** [comment]
Thank you for the guide and I will try to do my best about PRs

**Comment by MikePooh** [comment]
> right so link like (Server or WSGI):
> https://github.com/cutelyst/cutelyst/blob/master/server/CMakeLists.txt#L102
> and use it a bit like main.cpp of the same directory, and please do some PR to help me emit signals instead of qFatal or exit :)

I am successed embedding Cutelyst WSGI to my Qt-based application. Could you please give me more information regarding replacing qFatals and exit() to signals. I found lots of such calls, most of them are in unixfork.cpp and tcpserverbalancer.cpp.

I would make signals like example Q_SIGNAL void tcpListenError(const QString& err) end emit is each time this occurs. How do you think about this approach?

**Comment by dantti** [comment]
I think a singe errorOccured on Server class that in cutelyst**d** is bound to a qFatal to exit immediately (as that's usually what you want from a daemon that has failed)

**Comment by MikePooh** [comment]
I think the best way to explain is to make a minimal working example. I will try to do this within a few days.
